ocl-icd/ocl-icd-fix-build-error-with-gcc-10.patch

```diff
@@ -0,0 +1,41 @@
+From 4667bddd365bcc1dc66c483835971f0083b44b1d Mon Sep 17 00:00:00 2001
+From: Sergei Trofimovich <slyfox@gentoo.org>
+Date: Wed, 22 Jan 2020 19:38:23 +0000
+Subject: [PATCH] icd_generator.rb: fix build failure against gcc-10
+
+On gcc-10 (and gcc-9 -fno-common) build fails as:
+
+```
+libtool: link: gcc -shared  -fPIC -DPIC  .libs/libdummy_icd.o .libs/libdummy_icd_gen.o \
+    -ldl  -g -O2   -Wl,-soname -Wl,libdummycl.so.0 -o .libs/libdummycl.so.0.0.0
+ld: .libs/libdummy_icd_gen.o:/home/slyfox/dev/git/ocl-icd/libdummy_icd_gen.h:226:
+  multiple definition of `master_dispatch'; .libs/libdummy_icd.o:/home/slyfox/dev/git/ocl-icd/libdummy_icd_gen.h:226: first defined here
+```
+
+gcc-10 will change the default from -fcommon to fno-common:
+https://gcc.gnu.org/PR85678.
+
+The error also happens if CFLAGS=-fno-common passed explicitly.
+
+Reported-by: Anthony Parsons
+Bug: https://bugs.gentoo.org/706098
+Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
+---
+ icd_generator.rb | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/icd_generator.rb b/icd_generator.rb
+index d299dbd..6bd426d 100644
+--- a/icd_generator.rb
++++ b/icd_generator.rb
+@@ -189,7 +189,7 @@ EOF
+     }
+     libdummy_icd_structures += "};\n\n"
+     libdummy_icd_structures += "#pragma GCC visibility push(hidden)\n\n"
+-    libdummy_icd_structures += "struct _cl_icd_dispatch master_dispatch; \n\n"
++    libdummy_icd_structures += "extern struct _cl_icd_dispatch master_dispatch; \n\n"
```

Comment:
There is no need of this patch with 2.2.14 - https://github.com/OCL-dev/ocl-icd/blob/v2.2.14/icd_generator.rb#L211

Reply:
hi martin-g，I have a basic question to ask, here is to see whether the version has a patch through the source code, I went in to see how the specific is really whether there is a patch, please under your guidance.

Reply:
Hi @XingYuuuuu !

In this particular case I just checked the 2.2.14 version of `icd_generator.rb` file to see whether it contains the patch or not.
Usually the best is to find the commit that applies the patch and there you can see the first release with it.
https://github.com/OCL-dev/ocl-icd/commit/f83df609a3357314ddff2def6b7e7e2c5b93078c - here you can see that this commit has been made at May 14, 2020 and is part of [v2.3.2](https://github.com/OCL-dev/ocl-icd/releases/tag/v2.3.2) [v2.3.1](https://github.com/OCL-dev/ocl-icd/releases/tag/v2.3.1) [v2.3.0](https://github.com/OCL-dev/ocl-icd/releases/tag/v2.3.0) [v2.2.14](https://github.com/OCL-dev/ocl-icd/releases/tag/v2.2.14) [v2.2.13](https://github.com/OCL-dev/ocl-icd/releases/tag/v2.2.13)

Please re-phrase your question if I didn't understand you!